pkg/generate/app/errors.go

```diff
@@ -55,3 +49,7 @@ The argument %[1]q could apply to the following Docker images or OpenShift image
 %[2]s
 `, e.Image, buf.String())
 }
+
+// ErrNameRequired is the error returned by new-app when a name cannot be
+// suggested and the user needs to provide one explicitly.
+var ErrNameRequired = fmt.Errorf("you must specify a name for your output image with --name")
```

Comment:
It's not about an output image - it's "you must specify a name for your app with --name"